Async Python Docs (#277)
dbos-inc/dbos-docs: docs/python/tutorials/workflow-tutorial.md

All messages are persisted to the database, so if `send` completes successfully, the destination workflow is guaranteed to be able to `recv` it.
If you're sending a message from a workflow, DBOS guarantees exactly-once delivery because [workflows are reliable](./workflow-tutorial#reliability-guarantees).
If you're sending a message from normal Python code, you can use [`SetWorkflowID`](../reference/contexts.md#setworkflowid) with an idempotency key to guarantee exactly-once execution.

## Coroutine Workflows

> You can optionally define DBOS workflows as [coroutine functions](https://docs.python.org/3/glossary.html#term-coroutine-function). 

Reviewer: ```suggestion
DBOS workflows may be async functions (functions defined with `async def`, also called coroutines).
```

> These asyncronous workflows provide the same [reliability guarantees](#reliability-guarantees) as syncronous workflow functions. 

Reviewer: ```suggestion
These asynchronous workflows provide the same [reliability guarantees](#reliability-guarantees) as synchronous workflow functions. 
```

> Coroutine workflows provide the same [reliability guarantees](#reliability-guarantees) and [determinism](#determinism) as syncronous workflow functions. 

Reviewer: ```suggestion
Coroutine workflows provide the same [reliability guarantees](#reliability-guarantees) as synchronous workflow functions. 
```

Coroutine workflows may invoke [coroutine steps](./step-tutorial.md#coroutine-steps) via [await expressions](https://docs.python.org/3/reference/expressions.html#await).

> Additionally, coroutine workflows can and should use the asyncronous versions of the workflow [event](#workflow-events) and [messaging and notification](#workflow-messaging-and-notifications) context methods.

Reviewer: ```suggestion
Additionally, coroutine workflows can and should use the asynchronous versions of the workflow [event](#workflow-events) and [messaging and notification](#workflow-messaging-and-notifications) context methods.
```

:::info

> At this time, DBOS does not support coroutine [transactions](./transaction-tutorial.md). 

Reviewer: ```suggestion
At this time, DBOS does not support async [transactions](./transaction-tutorial.md). However, async workflows may safely call synchronous transactions.
```

Reviewer: Can async workflows safely call sync transactions?  If so, we should mention it.
